### PR TITLE
Experience/6930 Message Id Detail

### PR DESCRIPTION
**Pull request by penny-lischer**
This PR ...

Test Steps:
1. Add the "message-tracker" feature flag
2. Click on the Admin -> Message Id Search dropdown
3. Click search (no need for input since we are currently using mock data)
4. Click on a message Id in the table
5. View the message details

## Changes
- Added Message Id details including Sender, Warnings, Errors, Quality Filters
- Updated the feature flag name and renames files from "message-id-search" to "message-tracker"

Screen Shot 2022-10-18 at 12 53 01 PM

### Testing
- [x] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [X] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [X] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

### CODE REVIEW

File: frontend-react/src/components/MessageTracker/MessageDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+                className="grid-container margin-bottom-10"
+                data-testid="container"
+            >
+                <div className="margin-bottom-5">
+                    <h1>Message Details</h1>
+                    <Button onClick={() => navigate(-1)} type="button">
+                        Back
+                    </Button>
+                </div>
+                {withCatchAndSuspense(<MessageDetailsSender />)}
```
Comment (kevinhaube) [inline frontend-react/src/components/MessageTracker/MessageDetails.tsx:171]: Maybe a design question, but I'm curious if we'd want a spinner for each or one spinner until all data loads? Cc: @fmccaf1 

Regardless, I definitely think we'll want those catches on all of them!

File: frontend-react/src/components/MessageTracker/MessageTracker.tsx [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+// Main component.
+export function MessageTracker() {
+    const [searchFilter, setSearchFilter] = useState("");
+    const [isLoading, setIsLoading] = useState(false);
+    const [messagesData, setMessagesData] = useState<MessageListResource[]>([]);
+
+    const searchMessageId = async () => {
+        setIsLoading(true);
+
+        // TODO: make API call here to get data
```
Comment (kevinhaube) [inline frontend-react/src/components/MessageTracker/MessageTracker.tsx:102]: Is this for prototyping purposes, or are we going to need to fetch different from how we do with react-query?

File: frontend-react/src/components/MessageTracker/QualityFilters.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import React from "react";
+
+import { Destination } from "../../resources/ActionDetailsResource";
+
+//TODO: do we want to create a new interface here or use the existing Destination?
```
Comment (kevinhaube) [inline frontend-react/src/components/MessageTracker/QualityFilters.tsx:5]: Even if we stick with the old Destination, if it's something that's coming back from the API, I vote for moving it into `src/config/endpoints/[some file that makes sense to put it in].ts`. It'll make the cleanup of those rest-hooks Resource classes a lot easier when we come around to it 😃

File: frontend-react/src/components/MessageTracker/QualityFilters.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+export const QualityFilters = ({
+    qualityFilters,
+}: QualityFiltersDisplayProps) => {
+    return (
+        <>
+            <section className="margin-bottom-5">
+                <h3>Jurisdictions:</h3>
+                {qualityFilters?.map((d) => (
+                    <React.Fragment key={d.organization_id}>
```
Comment (kevinhaube) [inline frontend-react/src/components/MessageTracker/QualityFilters.tsx:18]: Today I learned you can use the `key` prop on a fragment. 😮

File: frontend-react/src/components/MessageTracker/WarningsErrors.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
+import React from "react";
+
+// TODO: move this interface into the resources directory
```
Comment (kevinhaube) [inline frontend-react/src/components/MessageTracker/WarningsErrors.tsx:3]: Doug had started moving network data interfaces like this into `src/config/endpoints/*.ts` where we store the endpoints they return from. There's a few examples in there already, but I think that Resources folder is heavily rest-hooks related so moving things we still use out of there helps with that cleanup later.

File: frontend-react/src/components/MessageTracker/WarningsErrors.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    title: string;
+    data: Warnings[];
+};
+
+export const WarningsErrors = ({ title, data }: WarningsErrorsDisplayProps) => {
+    return (
+        <>
+            <section className="margin-bottom-5">
+                <h3>{title}</h3>
+                <table
```
Comment (kevinhaube) [inline frontend-react/src/components/MessageTracker/WarningsErrors.tsx:21]: Is this something we could use the `Table` component for? The [DeliveryFacilitiesTable](https://github.com/CDCgov/prime-reportstream/blob/master/frontend-react/src/pages/deliveries/details/DeliveryFacilitiesTable.tsx) is a good example of a simpler table config.

**Review by kevinhaube — COMMENTED** [review]
First pass on the code, gonna take a poke around the UI in the morning. This looks great!

File: frontend-react/src/components/MessageTracker/MessageDetails.tsx (hunk 1 above)
Comment (penny-lischer) [reply frontend-react/src/components/MessageTracker/MessageDetails.tsx:171]: Good call, lets discuss.

File: frontend-react/src/components/MessageTracker/WarningsErrors.tsx (hunk 6 above)
Comment (penny-lischer) [reply frontend-react/src/components/MessageTracker/WarningsErrors.tsx:21]: I thought about it and didn't implement it for reasons. Will discuss with you and Aurora

Comment (kevinhaube) [reply frontend-react/src/components/MessageTracker/WarningsErrors.tsx:21]: Sounds good! 👍🏼

File: frontend-react/src/components/MessageTracker/QualityFilters.tsx (hunk 4 above)
Comment (kevinhaube) [reply frontend-react/src/components/MessageTracker/QualityFilters.tsx:18]: Super clutch -- I didn't know `<React.Fragment>` would render the same as `<>` so I've always had to wrap my fragments in a component and key the component instead. Glad I saw this so I don't have to anymore!

File: frontend-react/src/components/MessageTracker/MessageDetails.tsx (hunk 1 above)
Comment (kevinhaube) [reply frontend-react/src/components/MessageTracker/MessageDetails.tsx:171]: I reached out to Florence to get some input for that discussion as well, noting that neither really slow down our fetching process and either one can be its own useful experience (i.e. seeing data faster vs seeing more complete data all at once). I'll paste her reply here once I have it!

**Comment by penny-lischer** [comment]
@kevinhaube will make changes as part of #6931

**Review by ahay-agile6 — APPROVED** [review]

**Pull request merged**